### ISSUE TITLE
Investigate Latex support in comments again and make sure we can render it.

### ISSUE THREAD

**Issue by burtonator** [burtonator/polar-bookshelf#954]
I might need to setup just a BASIC demo of taking it as input, converting it to HTML , and saving it as html ...I wonder if I need to save it as the original Latex...

**Comment by devnoname120** [comment]
How would you allow editing a comment if you don't save the original LaTeX?

**Comment by burtonator** [comment]
yeah.. I guess this is a good point that we need to support editing the original latex.

**Comment by burtonator** [comment]
this is a good demo:

https://rawgit.com/tylerecouture/summernote-math/master/Example/example.html

it doesn't support updating. I think we would have to work on updates manually.

**Comment by burtonator** [comment]
That one supports editing. I think it keeps the latex as a hidden element in the HTML so you can re-edit it.

**Comment by rmnwnsx** [comment]
How about to use MatJax to render equations?
https://www.mathjax.org/

**Comment by rahuldeve** [comment]
Hi @burtonator Ive been using polar for the past few weeks and the only thing i wish it had was latex rendering. So I decided to see if I could add that feature on my own.

Since[ summernote-math](https://github.com/tylerecouture/summernote-math) didnt have the ability to edit, I decided to go ahead with just the [katex](https://katex.org/) library and an auto renderer [extension ](https://katex.org/docs/autorender.html)that is supported by katex. In the end I did get katex to render latex in polar although the way I made it work is a bit hacky. This is what Ive did for now:

- import katex and its auto renderer files in the index.html file for both pdf and html viewer
- use the `componentDidUpdate` hook on `AnnotationSidebar` component to trigger `renderMathInElement` function on the element having id as `annotaion-manager`. This renders any latex when viewing the annotation but does not when we edit it.

`renderMathInElement` is an API exposed by the auto renderer that iteratively renders latex within an element and its children. It looks for a set of configurable delimeters to see if latex is present or not.

katex requires `<!DOCTYPE html>` to function properly but adding this to the htmlviewer breaks scrolling in it. Adding `html, body { height: 100%; }` to the page seems to fix it for now.

I'm sure there is a proper way to use katex within polar but I dont have much experience with web development, hence this hacky way. I thought I'd let you be aware of a few gotchas when using katex.  Also I could get the current development branch to build in my system, otherwise I would have sent a PR for this :)
